homeassistant/components/denonavr/__init__.py

```diff
@@ -33,3 +43,54 @@ def service_handler(service):
         hass.services.register(DOMAIN, service, service_handler, schema=schema)
 
     return True
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up the denonavr components from a config entry."""
+    if hass.data.get(DOMAIN) is None:
+        hass.data[DOMAIN] = {}
+
+    # Connect to receiver
+    connect_denonavr = ConnectDenonAVR(
+        hass,
+        entry.data[CONF_HOST],
+        entry.data[CONF_TIMEOUT],
+        entry.data[CONF_SHOW_ALL_SOURCES],
+        entry.data[CONF_ZONE2],
+        entry.data[CONF_ZONE3],
+    )
+    await connect_denonavr.async_connect_receiver()
```

Comment:
`raise ConfigEntryNotReady` if we cannot cannot so its retried later

Reply:
thanks for the tip, done